app/src/main/java/pl/futurecollars/invoicing/service/TaxCalculatorService.java

```diff
@@ -15,36 +19,63 @@ public class TaxCalculatorService {
   private final Database database;
 
   public BigDecimal income(String taxIdentificationNumber) {
-    return database.visit(sellerPredicate(taxIdentificationNumber), InvoiceEntry::getPrice);
+    return database.visit(sellerPredicate(taxIdentificationNumber), InvoiceEntry::getNetPrice);
   }
 
   public BigDecimal costs(String taxIdentificationNumber) {
-    return database.visit(buyerPredicate(taxIdentificationNumber), InvoiceEntry::getPrice);
+    return database.visit(buyerPredicate(taxIdentificationNumber), this::getIncomeValueTakingIntoConsiderationPersonalCarUsage);
   }
 
-  public BigDecimal incomingVat(String taxIdentificationNumber) {
+  public BigDecimal collectedVat(String taxIdentificationNumber) {
     return database.visit(sellerPredicate(taxIdentificationNumber), InvoiceEntry::getVatValue);
   }
 
-  public BigDecimal outgoingVat(String taxIdentificationNumber) {
-    return database.visit(buyerPredicate(taxIdentificationNumber), InvoiceEntry::getVatValue);
+  public BigDecimal paidVat(String taxIdentificationNumber) {
+    return database.visit(buyerPredicate(taxIdentificationNumber), this::getVatValueTakingIntoConsiderationPersonalCarUsage);
   }
 
-  public BigDecimal getEarnings(String taxIdentificationNumber) {
-    return income(taxIdentificationNumber).subtract(costs(taxIdentificationNumber));
+  private BigDecimal getVatValueTakingIntoConsiderationPersonalCarUsage(InvoiceEntry invoiceEntry) {
+    return Optional.ofNullable(invoiceEntry.getExpenseRelatedToCar())
+        .map(Car::isPersonalUse)
+        .map(personalCarUsage -> personalCarUsage ? BigDecimal.valueOf(5, 1) : BigDecimal.ONE)
+        .map(proportion -> invoiceEntry.getVatValue().multiply(proportion))
+        .map(value -> value.setScale(2, RoundingMode.FLOOR))
+        .orElse(invoiceEntry.getVatValue());
+  }
+
+  private BigDecimal getIncomeValueTakingIntoConsiderationPersonalCarUsage(InvoiceEntry invoiceEntry) {
+    return invoiceEntry.getNetPrice()
+        .add(invoiceEntry.getVatValue())
+        .subtract(getVatValueTakingIntoConsiderationPersonalCarUsage(invoiceEntry));
   }
```

Comment:
Brakuje Ci tu jeszcze zaokrąglenia. Obecnie metoda zwróci np. wartość 391.5621213323232
dodaj jeszcze -  .setScale(2, RoundingMode.HALF_DOWN) (to wyjdzie dopiero w testach z MockMvc)